lib/vandelay/models/base.rb

```diff
@@ -9,20 +9,24 @@ def initialize(**data)
         data.each do |prop, val|
           self.instance_variable_set "@#{prop}", val
           self.class.class_eval do
+            # Tried to fix this by adding the following line, but it didn't work
+            # It caused problems in the specs. Values were getting cached between tests
+            # define_method("#{prop}") { val } unless method_defined?(prop)
             define_method("#{prop}") { val }
```

Comment:
This code causes a lot of ruby warnings:
```
/app/lib/vandelay/models/base.rb:15: warning: method redefined; discarding old vendor_id
/app/lib/vandelay/models/base.rb:15: warning: previous definition of vendor_id was here
```
I commented a potential fix, but it caused more problems. I think the root of the problem is because of the `class_eval`, refactoring to be regular methods would probably fix the issue. That would probably be solved better in a new PR.